Engine/EditorUI/Panel/InspectorSubpanel/PanelParticleSystem.cpp

```diff
@@ -224,26 +226,50 @@ void PanelParticleSystem::Render(ComponentParticleSystem* particle_system)
 		}
 
 		//Color of Particles
-
 		if (ImGui::CollapsingHeader("Color"))
 		{
 			particle_system->modified_by_user |= ImGui::ColorEdit4("Particle Color##2f", particle_system->initial_color.ptr(), ImGuiColorEditFlags_Float);
-			ImGui::Text("Fade Between Colors");
-			particle_system->modified_by_user |= ImGui::Checkbox("###Fade Between Colors", &particle_system->fade_between_colors);
-			ImGui::SameLine();
-			if (!particle_system->fade_between_colors)
+
+			ImGui::Spacing();
+			ImGui::Separator();
+			ImGui::Spacing();
+
+			int color_fade_type = static_cast<int>(particle_system->type_of_color_change);
+			if (ImGui::Combo("Fade Between Colors", &color_fade_type, "None\0Linear\0Curve\0"))
 			{
-				ImGui::PushItemFlag(ImGuiItemFlags_Disabled, true);
-				ImGui::PushStyleVar(ImGuiStyleVar_Alpha, ImGui::GetStyle().Alpha * 0.5f);
+				switch (color_fade_type)
```

Comment:
Because at the end of each case of this switch you are puting `particle_system->modified_by_user = true;`, you can simplify that by writing this line just after the switch